ga4gh/client.py

```diff
@@ -29,7 +54,89 @@ def __init__(self, urlPrefix, debugLevel, workarounds, key):
         self._workarounds = workarounds
         self._key = key
 
-    def runRequest(self, request, url, protocolClass, listAttr):
+    # TODO temporary auth solution
+    def _getAuth(self):
+        if self._usingWorkaroundsFor(self.workaroundGoogle):
+            return {'key': self._key}
+        else:
+            return {}
+
+    def _usingWorkaroundsFor(self, workaround):
+        return workaround in self._workarounds
+
+    def _debugResponse(self, jsonString):
+        if self._debugLevel > 1:
+            # TODO use a logging output and integrate with HTTP client more
+            # nicely... also, logging / error handling in general
+            print("json response:")
+            pp = json.dumps(
+                json.loads(jsonString), sort_keys=True, indent=4)
+            print(pp)
+
+    def _checkStatus(self, response):
+        if response.status_code != 200:
+            print(response.text)
+            raise Exception("Url {0} had status_code {1}".format(
+                response.url, response.status_code))
+
+    def _updateBytesRead(self, jsonString):
+        self._bytesRead += len(jsonString)
+
+    def _doRequest(self, requestContainer):
+        fullUrl = posixpath.join(
+            self._urlPrefix, requestContainer.url).format(
+                id=requestContainer.id_)
+        params = self._getAuth()
+        if requestContainer.requestMethod == 'post':
+            headers = {"Content-type": "application/json"}
+            jsonString = requestContainer.request.toJSONString()
+            response = requests.post(
+                fullUrl, jsonString, headers=headers, verify=False,
+                params=params)
+        elif requestContainer.requestMethod == 'get':
+            if requestContainer.request is not None:
+                requestDict = requestContainer.request.toJSONDict()
+                params.update(requestDict)
+            response = requests.get(fullUrl, params=params)
```

Comment:
We should have a final `else` here with an assert(False) in it, in case httpMethod is neither 'post' or 'get'. I'm a bit worried about these hard-coded strings being used for control flow. Can we used some constants maybe?